src/components/Inputs/AddressSelect.tsx

```diff
@@ -137,14 +136,7 @@ function AddressSelect({
             searchPlaceholder={t('Search for name or a hash...')}
             optionRender={(option) => {
               const address = addresses.find((address) => address.hash === option.value)
-              if (!address) return
-
-              return (
-                <SelectOptionItemContent
-                  ContentLeft={<AddressBadgeStyled addressHash={address.hash} showHashWhenNoLabel disableA11y />}
-                  ContentRight={<AmountStyled value={BigInt(address.balance)} fadeDecimals />}
-                />
-              )
+              if (address) return <SelectOptionAddress address={address} />
```

Comment:
Since we have more than 1 place where the address select option is useful (address select modal in Send modal and address select modal in Transfers page filtering), I extracted this to a new component. I did the same with the asset select option.